api/src/main/java/jakarta/servlet/http/HttpServlet.java

```diff
@@ -143,11 +143,7 @@ public HttpServlet() {
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
         String protocol = req.getProtocol();
         String msg = lStrings.getString("http.method_get_not_supported");
-        if (protocol.endsWith("1.1")) {
-            resp.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED, msg);
-        } else {
-            resp.sendError(HttpServletResponse.SC_BAD_REQUEST, msg);
-        }
+        switchProtocol(protocol, resp, msg);
```

Comment:
I don't like the method name... it is kind of unclear what it does.   I think the following approach would be better:
```suggestion
        resp.sendError(getMethodNotSupportedCode(protocol), msg);
```